### DISCUSSION TITLE
ADB over tcpip

### DISCUSSION THREAD

**Discussion by mikulski-lab** [DeviceFarmer/stf#704] [Q&A]
Hello everybody, I'm a a new user of STF.
My android device has got an external power supply. During my daily work I have to frequently import data from an USB drive and toggle between USB host/device mode. It's very annoying as I'm loosing access to STF.
Once STF installed is it possible to switch to permanently use adb over tcpip instead of the USB link?

**Comment by denis99999** [comment]
@mikulski-lab , yes but it is not so stable than USB link, for instance look at #700

**Reply by mikulski-lab** [reply]
Could you please guide me with parameters I shall use for SFT local? How to specify IP address of my android device? I would like to give it a try as I have got only a single android device connected.

**Comment by denis99999** [comment]
@mikulski-lab, firstly, using ADB is independent of STF, so you need to refer to the Android developer documentation for how to connect to a mobile phone using ADB over TCPIP through the WIFI or mobile network, or in any other way; secondly, once the mobile phone is seen as connected to your machine by ADB, you must launch STF using the `--allow-remote` option so that the mobile can be taken into account automatically by STF.

**Reply by mikulski-lab** [reply]
I'm launching everything from docker composer so I don't instrument adb directly. I know how to setup adb to reach my device over ip. Howerver I don't know how to setup the whole "docker" container to do that for me. By default the docker composer's adb runs in USB mode. I would like to switch to tcpip like I can do that from command line with >adb connect IP_ADDRESS:PORT.

**Comment by denis99999** [comment]
@mikulski-lab , your configuration and implementation problem is not related to STF, I already told you what to do at STF level (option `--allow-remote`), sorry but I can't help you further.
